### ISSUE TITLE
Internal compiler error when assigning to a Texture2D global

### ISSUE THREAD

**Issue by kayru** [microsoft/DirectXShaderCompiler#2989]
Here is a minimal repro shader, which is intentionally invalid. DXC crashes when attempting to report a compile error present in it.

```
Texture2D Textures[];
Texture2D MyTexture; // incorrect code that should report error, but crashes DXC
// static Texture2D MyTexture; // correct code that compiles
[numthreads(64,1,1)]
void main() { MyTexture = Textures[0]; }
```

**Comment by pow2clk** [comment]
I introduced this bug with the change to error messages. The fix should be pretty simple.

**Comment by pow2clk** [comment]
#3161 fixes the crash. This shader produces a lot of error messages, none of which are very helpful:

```
error: resource MyTexture could not be allocated Use /Zi for source location.
error: exported library functions cannot have resource parameters or return value. Value: ?MyTexture@@3V?$Texture2D@V?$vector@M$03@@@@A
Function: main: error: local resource not guaranteed to map to unique global resource. Use /Zi for source location.
```

Particularly unhelpful is "exported library functions cannot have resource parameters or return value". Whaaaaaaat? I'd like to fix these, but I think the priority is to fix the crash. So I'm fixing that now.
